ElectronicObserver/Window/Wpf/ShipTrainingPlanner/ShipTrainingPlanViewerViewModel.cs

```diff
@@ -0,0 +1,187 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel;
+using System.Linq;
+using CommunityToolkit.Mvvm.DependencyInjection;
+using CommunityToolkit.Mvvm.Input;
+using ElectronicObserver.Behaviors.PersistentColumns;
+using ElectronicObserver.Data;
+using ElectronicObserver.Database;
+using ElectronicObserver.Observer;
+using ElectronicObserver.Resource;
+using ElectronicObserver.ViewModels;
+using ElectronicObserver.Window.Dialog.ShipDataPicker;
+using ElectronicObserverTypes;
+using Jot;
+
+namespace ElectronicObserver.Window.Wpf.ShipTrainingPlanner;
+
+public partial class ShipTrainingPlanViewerViewModel : AnchorableViewModel
+{
+	private ShipDataPickerViewModel PickerViewModel { get; } = new();
+
+	public ObservableCollection<ShipTrainingPlanViewModel> Plans { get; set; } = new();
+
+	private ElectronicObserverContext DatabaseContext { get; } = new();
+
+	public List<ColumnProperties> ColumnProperties { get; set; } = new();
+	public List<SortDescription> SortDescriptions { get; set; } = new();
+
+	public ShipTrainingPlannerTranslationViewModel ShipTrainingPlanner { get; }
+
+	public ShipTrainingPlanViewModel? SelectedPlan { get; set; }
+
+	private Tracker Tracker { get; }
+
+	public ShipTrainingPlanViewerViewModel() : base("ShipTrainingPlanViewer", "ShipTrainingPlanViewer", ImageSourceIcons.GetIcon(IconContent.ItemActionReport))
+	{
+		Tracker = Ioc.Default.GetService<Tracker>()!;
+		ShipTrainingPlanner = Ioc.Default.GetRequiredService<ShipTrainingPlannerTranslationViewModel>();
+
+		Title = ShipTrainingPlanner.ViewTitle;
+		ShipTrainingPlanner.PropertyChanged += (_, _) => Title = ShipTrainingPlanner.ViewTitle;
+
+		SubscribeToApi();
+		StartJotTracking();
+	}
+
+	private void SubscribeToApi()
+	{
+		APIObserver o = APIObserver.Instance;
+
+		o.ApiPort_Port.ResponseReceived += (_, _) => Initialize();
+		o.ApiReqKousyou_DestroyShip.RequestReceived += OnShipScrap;
+	}
+
+	private void OnShipScrap(string apiname, dynamic data)
+	{
+		string idList = data["api_ship_id"].ToString();
+		IEnumerable<int> shipId = idList.Split(',').Select(int.Parse);
+
+		ShipTrainingPlanViewModel? planFound = Plans.FirstOrDefault(plan => shipId.Contains(plan.Ship.ID));
```

Comment:
Shouldn't this remove all plans, rather than just the first?

Reply:
having two plan for one ship should be a bug

Reply:
I mean, that shouldn't be possible to make at all. But you can scrap multiple ships at the same time, meaning shipId would be the ids for all the ships you scrapped. So if you had plans for 2 different ships and scrapped them both, you should remove both plans. The current logic would only remove 1 plan.